airflow/executors/knative_executor.py

```diff
@@ -0,0 +1,186 @@
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+import multiprocessing
+
+from cached_property import cached_property
+from airflow.executors.base_executor import BaseExecutor
+from airflow.configuration import conf
+from airflow.utils.log.logging_mixin import LoggingMixin
+from airflow.utils.state import State
+from airflow.exceptions import AirflowException
+import asyncio
+import datetime
+import aiohttp
+import functools
+from airflow.contrib.executors.kubernetes_executor import KubernetesExecutor
+
+
+async def make_request_async(
+    task_id,
+    dag_id,
+    execution_date,
+    host,
+    log,
+    host_header=None,
+) -> aiohttp.ClientResponse:
+    req = "http://" + host + "/run"
+
+    date = int(datetime.datetime.timestamp(execution_date))
+    params = {
+        "task_id": task_id,
+        "dag_id": dag_id,
+        "execution_date": date,
+    }
+    headers = {}
+    if host_header:
+        headers["Host"] = host_header
+    timeout = aiohttp.ClientTimeout(total=60000)
+    async with aiohttp.ClientSession(timeout=timeout) as session:
+        async with session.get(url=req, params=params, headers=headers) as resp:
+            log.info(resp.status)
+            log.info(await resp.text())
+            return resp
+
+
+class KnativeRequestLoop(multiprocessing.Process, LoggingMixin):
+    def __init__(self,
+                 task_pipe,
+                 result_pipe,
+                 pop_running_pipe,
+                 host_header,
+                 host,
+                 ):
+
+        super().__init__()
+        self.host = host
+        self.task_pipe = task_pipe
+        self.result_pipe = result_pipe
+        self.host_header = host_header
+        self.pop_running_pipe = pop_running_pipe
+
+    def recieve_and_execute(self, loop):
+        key = self.task_pipe.recv()
+        loop.create_task(self.execute_work(key=key))
+
+    def run(self):
+        loop = asyncio.get_event_loop()
+        loop.add_reader(fd=self.task_pipe, callback=functools.partial(self.recieve_and_execute, loop))
+        loop.run_forever()
+
+    async def execute_work(self, key):
+        """
+        Executes command received and stores result state in queue.
+        :param task_instance:
+        :param key: the key to identify the TI
+        :type key: tuple(dag_id, task_id, execution_date)
```

Comment:
```suggestion
        :type key: tuple(dag_id, task_id, execution_date, try_number)
```

but those aren't types ;)

Reply:
Yup, if adding type hints: `Tuple(str,str,str,str)` I think

Reply:
The local executor has the same docstring that needs to be corrected:

https://github.com/astronomer/airflow/blob/master/airflow/executors/local_executor.py#L77